x/ibc/light-clients/solomachine/types/proposal_handle.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/exported"
+)
+
+// CheckProposedHeaderAndUpdateState updates the consensus state to the header's sequence and
+// public key. An error is returned if the header cannot be casted to a solo machine header.
+func (cs ClientState) CheckProposedHeaderAndUpdateState(
```

Comment:
pending test

Reply:
yup on the todo to add tests for this func